src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -998,6 +998,11 @@ memory::ptr primitive_inst::allocate_internal_buffer(size_t idx, bool reset) {
         alloc_type = engine.get_lockable_preferred_memory_allocation_type();
     }
     GPU_DEBUG_LOG << "=> allocate to " << alloc_type << std::endl;
+
+    // Reuse intermediate buffer like output buffer.
+    if (_node->get_preferred_impl_type() == impl_types::onednn) {
+        return _network.get_memory_pool().get_intermediate_memory(layout, _node->id(), _network.get_id(), _node->get_memory_dependencies(), alloc_type);
```

Comment:
why do we treat onednn case in a special way compared to cldnn primitives?

Reply:
It has changed.